consensus/consensus-types/src/block.rs

```diff
@@ -210,7 +227,7 @@ where
     /// If this is the genesis block, we skip these checks.
     pub fn validate_signatures(&self, validator: &ValidatorVerifier) -> failure::Result<()> {
```

Comment:
could you add a TODO into safety rules about verifying genesis ledger infos, since we won't verify them in the same way we verify existing LIs / QCs? Or let's chat today so we can scope out what this might look like. I suspect as you have already mentioned, we'll likely need a special API that takes in the last QC of an epoch and computes internally the next state.

Reply:
I thought safety rules would use the same verify functions but just do double-check inside?